nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/ListSmb.java

```diff
@@ -0,0 +1,353 @@
+                + "continuously without duplication."
+)
+public class ListSmb extends AbstractListProcessor<SmbListableEntity> {
+
+    public static final PropertyDescriptor DIRECTORY = new PropertyDescriptor.Builder()
+            .displayName("Input Directory")
+            .name("directory")
+            .description("The network folder to which files should be written. This is the remaining relative " +
+                    "after the hostname: smb://HOSTNAME:PORT/SHARE/[DIRECTORY]\\sub\\directories. It is also possible "
+                    + " to add subdirectories using this property. The given path on the remote file share must exists. "
+                    + "The existence of the remote folder can be checked using verification. You may mix different "
+                    + "directory separators in this property. If so NiFi will unify all of them and will use windows's"
+                    + "directory separator: '\\' ")
+            .required(false)
+            .addValidator(NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MINIMUM_AGE = new PropertyDescriptor.Builder()
+            .displayName("Minimum File Age in milliseconds")
+            .name("min-file-age")
+            .description(
+                    "Any file younger then the given value will be omitted. Ideally this value should be greater then"
+                            + "the amount of time needed to perform a list.")
+            .required(true)
+            .addValidator(TIME_PERIOD_VALIDATOR)
+            .defaultValue("5ms")
+            .build();
+
+    public static final PropertyDescriptor MINIMUM_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Minimum File Size in bytes")
+            .name("min-file-size")
+            .description("Any file smaller then the given value will be omitted.")
+            .required(false)
+            .addValidator(NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MAXIMUM_SIZE = new PropertyDescriptor.Builder()
+            .displayName("Maximum File Size in bytes")
+            .name("max-file-size")
+            .description("Any file bigger then the given value will be omitted.")
+            .required(false)
+            .addValidator(NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+
+    public static final PropertyDescriptor SMB_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(LISTING_STRATEGY)
+            .allowableValues(BY_ENTITIES, NO_TRACKING, BY_TIMESTAMPS)
+            .build();
+
+    public static final PropertyDescriptor SMB_CONNECTION_POOL_SERVICE = new Builder()
+            .name("smb-connection-pool-service")
+            .displayName("SMB Connection Pool Service")
+            .description("Specifies the SMB Connection Pool to use for creating SMB connections.")
```

Comment:
This should be adjusted based on the interface renaming:
```suggestion
            .name("smb-session-provider-service")
            .displayName("SMB Session Provider Service")
            .description("Specifies the SMB Session Provider to use for obtaining SMB connections.")
```